src/aspire/utils/relion_interop.py

```diff
@@ -50,42 +50,130 @@
 
 
 def df_to_relion_types(df):
-    # convert STAR file strings to data type for each field
-    # columns without a specified data type are read as dtype=object
+    """
+    Convert STAR file strings to data type for each field in a DataFrame loaded via
+    `aspire.storage.StarFile`. Columns without a specified data type are read as
+    `dtype=object`.
+
+    :param df: A `StarFile` block represented as a Pandas DataFrame.
+    :return: The data frame with types converted where possible.
+    """
     column_types = {name: relion_metadata_fields.get(name, str) for name in df.columns}
     return df.astype(column_types)
 
 
-class Relion30StarFile(StarFile):
+class RelionStarFile(StarFile):
+    """
+    A star file generated by RELION representing particles, micrographs, or movies.
+    """
+
     def __init__(self, filepath):
         super().__init__(filepath, blocks=None)
 
-        # first convert types where possible
+        # convert dtypes in star file blocks where possible
+        self.convert_dtypes()
+
+        # validate star file and detect Relion version (3.0 or >=3.1)
+        self.validate_and_detect_version()
+
+    def validate_and_detect_version(self):
```

Comment:
This and `convert_dtypes` are only ever called internally, so maybe we should make them private?